### PR TITLE
Add Support to run test over BAREMETAL

### PR DESCRIPTION
**Pull request by prsurve**
Signed-off-by: Pratik Surve <pratiksurve2130@gmail.com>

### CODE REVIEW

File: ocs_ci/deployment/baremetal.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class BAREMETALUPI:
+    """
+    A class to handle vSphere UPI specific deployment
```
Comment (RazTamir) [inline ocs_ci/deployment/baremetal.py:8]: Bare metal instead of vsphere

**Comment by vasukulkarni** [comment]
LGTM, fix the one pending comment

**Comment by prsurve** [comment]
Failed tests/manage/z_cluster/cluster_expansion/test_add_capacity.py::TestAddCapacity::test_add_capacity 
Skipped
tests/e2e/logging/test_openshift-logging.py::Testopenshiftloggingonocs::test_create_new_project_to_verify_logging::setup

tests/manage/mcg/test_multi_region.py::TestMultiRegion::test_multiregion_bucket_creation::setup

tests/manage/z_cluster/cluster_expansion/test_node_expansion.py::TestAddNode::test_add_node

tests/manage/z_cluster/nodes/test_nodes_maintenance.py::TestNodesMaintenance::test_node_maintenance[worker]::setup

tests/manage/z_cluster/nodes/test_nodes_maintenance.py::TestNodesMaintenance::test_node_maintenance[master]::setup

rest of tier1 testcase passed

**Comment by ebenahar** [comment]
> rest of tier1 test case passed

@prsurve verified?

**Comment by prsurve** [comment]
> > rest of tier1 test case passed
> 
> @prsurve verified?

Tier 1 run passed i have scheduled tier4a run to make sure everything work fine as we get expected  result if API for node restart are missing

**Comment by prsurve** [comment]
Tier4a failed test  


Test Name | Duration | Age
-- | -- | --
tests.e2e.scale.test_scale_osds_fill_75%_reboot_workers.TestScaleOSDsRebootNodes.test_scale_osds_reboot_nodes[CephBlockPool] | 13 hr | 1
tests.e2e.scale.test_scale_osds_fill_75%_reboot_workers.TestScaleOSDsRebootNodes.test_scale_osds_reboot_nodes[CephBlockPool] | 1.1 sec | 1
tests.e2e.scale.test_scale_osds_fill_75%_reboot_workers.TestScaleOSDsRebootNodes.test_scale_osds_reboot_nodes[CephFileSystem] | 0.36 sec | 1
tests.manage.monitoring.prometheus.test_deployment_status.test_ceph_manager_stopped | 0.35 sec | 1
tests.manage.monitoring.prometheus.test_deployment_status.test_ceph_monitor_stopped | 0.35 sec | 1
tests.manage.monitoring.prometheus.test_deployment_status.test_ceph_osd_stopped | 0.94 sec | 1
tests.manage.monitoring.prometheus.test_noobaa.test_noobaa_bucket_quota | 54 sec | 1
tests.manage.pv_services.test_pvc_disruptive.TestPVCDisruption.test_pvc_disruptive[CephBlockPool-run_io-operator] | 14 min | 1
tests.manage.z_cluster.nodes.test_nodes_restart.TestNodesRestart.test_nodes_restart[True] | 7.9 sec | 1
tests.manage.z_cluster.nodes.test_nodes_restart.TestNodesRestart.test_rolling_nodes_restart | 9.3 sec | 1
tests.manage.z_cluster.nodes.test_nodes_restart.TestNodesRestart.test_pv_provisioning_under_degraded_state_stop_provisioner_pod_node[rbd-create_resources] | 8.9 sec | 1
tests.manage.z_cluster.nodes.test_nodes_restart.TestNodesRestart.test_pv_provisioning_under_degraded_state_stop_provisioner_pod_node[rbd-delete_resources] | 8.3 sec | 1
tests.manage.z_cluster.nodes.test_nodes_restart.TestNodesRestart.test_pv_provisioning_under_degraded_state_stop_provisioner_pod_node[cephfs-create_resources] | 7.9 sec | 1
tests.manage.z_cluster.nodes.test_nodes_restart.TestNodesRestart.test_pv_provisioning_under_degraded_state_stop_provisioner_pod_node[cephfs-delete_resources] | 8.1 sec | 1
tests.manage.z_cluster.nodes.test_nodes_restart.TestNodesRestart.test_pv_provisioning_under_degraded_state_stop_rook_operator_pod_node[create_resources] | 8.2 sec | 1
tests.manage.z_cluster.nodes.test_nodes_restart.TestNodesRestart.test_pv_provisioning_under_degraded_state_stop_rook_operator_pod_node[delete_resources] | 8.4 sec | 1


Test which were doing restart failed with reason 
`
@pytest.fixture()
    def nodes():
        """
        Return an instance of the relevant platform nodes class
        (e.g. AWSNodes, VMWareNodes) to be later used in the test
        for nodes related operations, like nodes restart,
        detach/attach volume, etc.
    
        """
        factory = platform_nodes.PlatformNodesFactory()
>       nodes = factory.get_nodes_platform()

tests/conftest.py:1613: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ocs_ci.ocs.platform_nodes.PlatformNodesFactory object at 0x7f25d6c1b208>

    def get_nodes_platform(self):
        platform = config.ENV_DATA['platform']
>       return self.cls_map[platform]()
E       KeyError: 'baremetal'
` 
with commit 512ae05aaa46d41493d461f0703d12f483b32281 it will fix the issue we will get proper error saying 
self = <ocs_ci.ocs.platform_nodes.NodesBase object at 0x7f9becd61cd0>

    def restart_nodes_teardown(self):
        raise NotImplementedError(
>           "Restart nodes teardown functionality is not implemented"
        )
E       NotImplementedError: Restart nodes teardown functionality is not implemented

ocs_ci/ocs/platform_nodes.py:93: NotImplementedError

File: ocs_ci/ocs/platform_nodes.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ class PlatformNodesFactory:
 
     """
     def __init__(self):
-        self.cls_map = {'AWS': AWSNodes, 'vsphere': VMWareNodes, 'aws': AWSNodes}
+        self.cls_map = {'AWS': AWSNodes, 'vsphere': VMWareNodes, 'aws': AWSNodes, 'baremetal': NodesBase}
```
Comment (ebenahar) [inline ocs_ci/ocs/platform_nodes.py:31]: @prsurve need to create a new class for baremetal here. Ok with doing that in a separate PR. Please just report an issue to track it

Comment (prsurve) [reply ocs_ci/ocs/platform_nodes.py:31]: Done create an issue #1965

Comment (ebenahar) [reply ocs_ci/ocs/platform_nodes.py:31]: thanks

**Pull request merged**